codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/smoketests/SmokeTestsRunnerGenerator.kt

```diff
@@ -0,0 +1,174 @@
+import software.amazon.smithy.smoketests.traits.SmokeTestsTrait
+import kotlin.jvm.optionals.getOrNull
+
+/**
+ * Renders smoke tests runner for a service
+ */
+class SmokeTestsRunnerGenerator(
+    private val writer: KotlinWriter,
+    private val service: Symbol,
+    private val operations: List<OperationShape>,
+    private val model: Model,
+    private val symbolProvider: SymbolProvider,
+    private val sdkId: String,
+) {
+    internal fun render() {
+        writer.write("import kotlin.system.exitProcess")
+        writer.emptyLine()
+        writer.write("private var exitCode = 0")
+        writer.write("private val regionOverride = System.getenv(\"AWS_SMOKE_TEST_REGION\")")
+        writer.write("private val skipTags = System.getenv(\"AWS_SMOKE_TEST_SKIP_TAGS\")?.let { it.split(\",\").map { it.trim() }.toSet() }")
+        writer.emptyLine()
+        writer.withBlock("public suspend fun main() {", "}") {
+            renderFunctionCalls()
+            write("exitProcess(exitCode)")
+        }
+        writer.emptyLine()
+        renderFunctions()
+    }
+
+    private fun renderFunctionCalls() {
+        operations.forEach { operation ->
+            operation.getTrait<SmokeTestsTrait>()?.testCases?.forEach { testCase ->
+                writer.write("${testCase.id.toCamelCase()}()")
+            }
+        }
+    }
+
+    private fun renderFunctions() {
+        operations.forEach { operation ->
+            operation.getTrait<SmokeTestsTrait>()?.testCases?.forEach { testCase ->
+                renderFunction(operation, testCase)
+                writer.emptyLine()
+            }
+        }
+    }
+
+    private fun renderFunction(operation: OperationShape, testCase: SmokeTestCase) {
+        writer.withBlock("private suspend fun ${testCase.id.toCamelCase()}() {", "}") {
+            write("val tags = setOf<String>(${testCase.tags.joinToString(",") { it.dq()} })")
+            write("if (skipTags != null && tags.any { it in skipTags }) return")
+            emptyLine()
+            withInlineBlock("try {", "} ") {
+                renderClient(testCase)
+                renderOperation(operation, testCase)
+            }
+            withBlock("catch (e: Exception) {", "}") {
+                renderCatchBlock(testCase)
+            }
+        }
+    }
+
+    private fun renderClient(testCase: SmokeTestCase) {
+        writer.withInlineBlock("#L {", "}", service) {
+            if (testCase.vendorParams.isPresent) {
+                testCase.vendorParams.get().members.forEach { vendorParam ->
+                    if (vendorParam.key.value == "region") {
+                        write("region = regionOverride ?: #L", vendorParam.value.render())
+                    } else {
+                        write("#L = #L", vendorParam.key.value.toCamelCase(), vendorParam.value.render())
+                    }
+                }
+            } else {
+                write("region = regionOverride")
```

Comment:
What happens when the `AWS_SMOKE_TEST_REGION` is not set?

Reply:
In that case `regionOverride ` will be set to `null`